src/test/java/uk/gov/hmcts/reform/cwrdapi/controllers/StaffRefDataControllerTest.java

```diff
@@ -152,6 +217,379 @@ void shouldFetchEmptyUserTypes() {
+        searchResponse.setSkills(skills);
+        searchReq = SearchRequest.builder()
+                .skill("1")
+                .build();
+
+        advancedSearchResponse = new ResponseEntity<>(List.of(searchResponse), null, HttpStatus.OK);
+        when(staffRefDataService.retrieveStaffProfile(eq(searchReq), Mockito.any(PageRequest.class)))
+                .thenReturn(advancedSearchResponse);
+        ResponseEntity<List<SearchStaffUserResponse>> response =
+                staffRefDataController.searchStaffProfile(1,2, searchReq);
+        assertNotNull(response);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        verify(staffRefDataService, times(1))
+                .retrieveStaffProfile(eq(searchReq), Mockito.any(PageRequest.class));
+
+        assertTrue(validateSearchUserProfileResponse(response,searchReq));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"avdx","_*kd"})
+    void should_return_staffProfiles_search_with_invalid_skillId_status_code_400(String skillId) {
+        searchReq = SearchRequest.builder()
+                .skill(skillId)
+                .build();
+
+        assertThrows(InvalidRequestException.class, () -> staffRefDataController
+                .searchStaffProfile(2,2, searchReq));
+        verify(staffRefDataService, times(0))
+                .retrieveStaffProfile(eq(searchReq), Mockito.any(PageRequest.class));
+    }
+
+
+
+    @ParameterizedTest
+    @ValueSource(strings = {"task supervisor,case allocator,staff administrator"})
+    void should_return_staffProfiles_search_with_role_status_code_200(String roleId) {
+        searchResponse.setTaskSupervisor(true);
+        searchResponse.setCaseAllocator(true);
+        searchResponse.setStaffAdmin(true);
+
+        searchReq = SearchRequest.builder()
+                .role(roleId)
+                .build();
+
+        advancedSearchResponse = new ResponseEntity<>(List.of(searchResponse), null, HttpStatus.OK);
+        when(staffRefDataService.retrieveStaffProfile(eq(searchReq), Mockito.any(PageRequest.class)))
+                .thenReturn(advancedSearchResponse);
+        ResponseEntity<List<SearchStaffUserResponse>> response = staffRefDataController
+                .searchStaffProfile(1,2, searchReq);
+        assertNotNull(response);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        verify(staffRefDataService, times(1))
+                .retrieveStaffProfile(eq(searchReq), Mockito.any(PageRequest.class));
+
+        assertTrue(validateSearchUserProfileResponse(response,searchReq));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"tasksupervisor,case_allocator,abcd"})
+    void should_return_staffProfiles_search_with_invalid_role_status_code_400(String roleId) {
+
+        searchReq = SearchRequest.builder()
+                .role(roleId)
+                .build();
+
+        assertThrows(InvalidRequestException.class, () -> staffRefDataController
```

Comment:
can you check for expected error message

Reply:
changes done